gamification/process/update.py

```diff
@@ -135,6 +137,38 @@ def get_participant_objects():
     return participants
 
 
+def get_meta_review_participant_objects():
+    return meta_review_participants.objects.all()
+
+
+def update_participants_data_with_meta_review(meta_review_participant):
+    """
+    Update participants based on meta-review
+
+    This method updates every participant based on the meta-review
+    received or given. If the gamification participant is
+    in the active newcomers list then it will check if the meta-review
+    is complete or not and update the activity accordingly.
+    """
+    active_newcomers_list = active_newcomers()
+    if meta_review_participant.login in active_newcomers_list:
+        gamification_participant = Participant.objects.get(
+            username=meta_review_participant.login)
+        reaction_name = Reaction.objects.get(id=meta_review_participant.login)
+        if (meta_review_participant.pos_in > 0 or
+                meta_review_participant.pos_out > 0 or
+                meta_review_participant.neg_in > 0 or
+                meta_review_participant.neg_out > 0):
+            """
+            Add a meta-review activity to the participant
+            """
+            activity = 'Did a meta-review or received a meta-review'
+            created_at = reaction_name.created_at
+            updated_at = None
+            gamification_participant.add_activity(0, activity, created_at,
+                                                  updated_at)
```

Comment:
I am not 100% sure here, but I suspect this would add repetitive activities to newcomers, because you don't check whether the newcomer has already completed a meta-review activity.

Reply:
Also add a testcase to verify that ;)
especially if you believe I am wrong here.

Reply:
> I suspect this would add repetitive activities to newcomers

Sounds logical but in order to do this, I will have to add in a flag to ensure if the meta-review has been completed. 
Or should I create another list for the participants who have completed the meta-review. While passing on a participant's name, we can check if he's in the list or not. In case it is, we can break the statement.

Reply:
Maybe before adding a new activity, check if the user has already finished a meta-review activity. I’m not sure how gamification app avoids other duplicated activities, maybe you can learn from them.